sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3701,6 +3703,16 @@ object CleanupAliases extends Rule[LogicalPlan] with AliasHelper {
   }
 }
 
+/**
+ * A rule that calls `TransformationAfterAnalysis.transform` for a plan that needs to be transformed
+ * after all the analysis rules are run.
+ */
+object TransformAfterAnalysis extends Rule[LogicalPlan] {
+  override def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
```

Comment:
Technically, this can be just `plan match`, but used `resolveOperators` to be future proof. Please let me know if this is an overkill.